Modules/CustomerHistory/Frames.lua

```diff
@@ -24,15 +24,46 @@ function CraftSim.CUSTOMER_HISTORY.FRAMES:Init()
     local function createContent(frame)
         self.frame:Hide()
         self.frame.content.customerDropdown = CraftSim.GGUI.Dropdown({
-            parent=self.frame.content, anchorParent=self.frame.title.frame, anchorA="TOP", anchorB="TOP", offsetY=-30, width=170, offsetX=-170,
+            parent=self.frame.content, anchorParent=self.frame.title.frame, anchorA="TOP", anchorB="TOP", width=170, offsetX=-170, offsetY=-30,
             initialValue=nil,
             initialLabel="",
             label=CraftSim.LOCAL:GetText(CraftSim.CONST.TEXT.CUSTOMER_HISTORY_DROPDOWN_LABEL),
             initialData={},
             clickCallback=function (_, _, item) CraftSim.CUSTOMER_HISTORY.FRAMES:SetCustomer(item) end
         })
+        self.frame.content.deleteButton = CraftSim.GGUI.Button({
+            parent=frame.content,anchorParent=self.frame.content.customerDropdown.frame, anchorA="LEFT", anchorB="RIGHT", offsetX=0, offsetY=2,
+            label=CraftSim.LOCAL:GetText(CraftSim.CONST.TEXT.CUSTOMER_HISTORY_DELETE_BUTTON), sizeX=15, sizeY=20, adjustWidth=true,
+            clickCallback=function ()
+                CraftSim.CUSTOMER_HISTORY.db.realm[CraftSim.CUSTOMER_HISTORY.db.realm.lastCustomer] = nil
+                self:ResetDropdown()
+                self.frame.content.messageBox:Clear()
+                self.frame.content.totalTip:SetText(string.format("%s%s", CraftSim.LOCAL:GetText(CraftSim.CONST.TEXT.CUSTOMER_HISTORY_TOTAL_TIP), CraftSim.GUTIL:FormatMoney(0)))
+                CraftSim.CUSTOMER_HISTORY.db.realm.lastCustomer = nil
+                self.frame.content.deleteButton.frame:SetEnabled(false)
+            end,
+        })
+        self.frame.content.deleteButton.frame:SetEnabled(false)
+
+        self.frame.content.filterTextbox = CraftSim.FRAME:CreateInput("Search", self.frame.content, self.frame.content.customerDropdown.frame, "TOP", "BOTTOM", 0, 0, 170, 25,
+        "",
+        function(textInput, userInput)
+            if userInput then
+                input = textInput:GetText() or ""
+                self:ResetDropdown(CraftSim.CUSTOMER_HISTORY.db.realm.lastCustomer, input)
+                if not _G["DropDownList"..1]:IsShown() then
+                    ToggleDropDownMenu(1, nil, self.frame.content.customerDropdown.frame, self.frame.content.customerDropdown.frame.name, 170, 50, nil, nil)
+                end
+            else
+                self:ResetDropdown(CraftSim.CUSTOMER_HISTORY.db.realm.lastCustomer)
+                if _G["DropDownList"..1]:IsShown() then
+                    ToggleDropDownMenu(1, nil, self.frame.content.customerDropdown.frame, CraftSim.CUSTOMER_HISTORY.FRAMES.frame.content.customerDropdown.frame.name, 0, 0, nil, nil)
```

Comment:
If the dropdown is shown then show it? I am not sure to understand this part.

Reply:
This is a weird trick. The only way I found to Show or Hide a dropdown was to use ToggleDropDownMenu.
The idea was that if the user input is nil or empty, it would Toggle the dropdown to actually hide it. The same way above it only toggles to show it if it is not shown already

Reply:
The `userInput` is actually a boolean, meaning that it was triggered by the user's interaction. As far as I know, this is only triggered when the box is initialized.